CadRevealComposer/Operations/SectorSplitting/PrioritySectorSplitter.cs

```diff
@@ -0,0 +1,89 @@
+﻿namespace CadRevealComposer.Operations.SectorSplitting;
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using IdProviders;
+using Primitives;
+
+public class PrioritySectorSplitter : ISectorSplitter
+{
+    private const long SectorEstimatedByteSizeBudget = 50_000; // bytes, Arbitrary value
+
+    public IEnumerable<InternalSector> SplitIntoSectors(
+        APrimitive[] allGeometries,
+        SequentialIdGenerator sectorIdGenerator
+    )
+    {
+        if (sectorIdGenerator.PeekNextId == 0)
+            _ = sectorIdGenerator.GetNextId(); // Get and discard id 0, as 0 is hardcoded below
+        var dummyRootSector = SplittingUtils.CreateRootSector(0, "/0", new BoundingBox(Vector3.Zero, Vector3.One));
+        yield return dummyRootSector; // Using dummy root sector because we need to remap it to the non-priority sector root later on
+
+        var sectors = new List<InternalSector>();
+        // We split the geometries into sectors based on the discipline because there are very few highlight cases that are cross-discipline
+        // This helps reduce the overhead of a priority sector, since highlighting of multiple tags within the same discipline is more common
+        var primitivesGroupedByDiscipline = allGeometries.GroupBy(x => x.Discipline);
+        foreach (var disciplineGroup in primitivesGroupedByDiscipline)
+        {
+            var geometryGroups = disciplineGroup.GroupBy(primitive => primitive.TreeIndex); // Group by treeindex to avoid having one treeindex unnecessary many sectors
+            var nodes = PrioritySplittingUtils.ConvertPrimitiveGroupsToNodes(geometryGroups);
+
+            // Ignore outlier nodes
+            // TODO: Decide if this is the right thing to do
```

Comment:
Not sure about this one, since there might be exceptions where outlier islands actually are parts of the asset??? In most cases, though, it is reasonable to exclude outliers

Reply:
Log how many primitives was considered outliers